docs/source/trainer/using_the_trainer.rst

```diff
@@ -0,0 +1,505 @@
+or ``T_max`` can be provided in many units, from epochs (``"10ep"``)
+to batches (``"2048ba"``) to duration (``"0.7dur"``). See
+:doc:`Schedulers`<optimizers_and_schedulers> for details.
+
+
+Training on GPU
+~~~~~~~~~~~~~~~
+
+Control which device you use for training with the ``device`` parameter,
+and we will handle the data movement and other systems-related
+engineering. We currently support the ``cpu`` and ``gpu`` devices.
+
+.. code:: python
+
+   from composer import Trainer
+
+   trainer = Trainer(model=model,
+                     train_dataloader=train_dataloader,
+                     eval_dataloader=eval_dataloader,
+                     max_duration='160ep',
+                     device='gpu')
+
+Distributed Training
+~~~~~~~~~~~~~~~~~~~~
+
+It's also simple to do data-parallel training on multiple GPUs. Composer
+provides a launcher command that works with the trainer and handles all
+the ``torch.distributed`` setup for you.
+
+.. code:: python
+
+   # run_trainer.py
+
+   from composer import Trainer
+
+   trainer = Trainer(model=model,
+                     train_dataloader=train_dataloader,
+                     eval_dataloader=eval_dataloader,
+                     max_duration='160ep',
+                     device='gpu')
+   trainer.fit()
+
+Access the Composer launcher via the ``composer`` command along with the
+number of GPUs you'd like to use and your training script. Use
+``composer --help`` to see a full list of configurable options.
+
+.. code:: bash
+
+   # run training on 8 GPUs
+   $ composer -n 8 run_trainer.py
+
+For multiple GPUs, the ``batch_size`` for each dataloader should be the
+per-device batch size. For example, to use a batch size of 2048, with
+data parallel across 8 GPUs, the dataloader should have ``batch_size=256``.
+
+
+.. warning::
+
+    If using distributed training, your dataloader use the
+    :mod:`torch.utils.data.distributed.DistributedSampler`.
+
+
+.. seealso::
+
+    Our :doc:`Distributed Training<distributed_training>` guide and
+    the :mod:`composer.utils.dist` module.
+
+
+DeepSpeed Integration
+~~~~~~~~~~~~~~~~~~~~~
+
+Composer comes with DeepSpeed support, allowing you to leverage their
+full set of features that makes it easier to train large models across
+(1) any type of GPU and (2) multiple nodes. For more details on DeepSpeed,
+see `their website <https://www.deepspeed.ai>`__.
+
+To enable DeepSpeed, simply pass in a config as specified in the
+DeepSpeed docs `here <https://www.deepspeed.ai/docs/config-json/>`__.
+
+.. code:: python
+
+   # run_trainer.py
+
+   from composer import Trainer
+
+   trainer = Trainer(model=model,
+                     train_dataloader=train_dataloader,
+                     eval_dataloader=eval_dataloader,
+                     max_duration='160ep',
+                     device='gpu',
+                     deepspeed_config={
+                         "train_batch_size": 2048,
+                         "amp": {"enabled": True},
```

Comment:
Note: According to @jbloxham "amp" doesn't really work this way as documented by DeepSpeed. It would be better to change the to 

```
"fp16": {"enabled": True}
```

Reply:
fixed!